### PR TITLE
RHCLOUD-23647 Introduce emails only mode

### PR DESCRIPTION
**Pull request by gwenneg**
From now on, if `NOTIFICATIONS_EMAILS_ONLY_MODE_ENABLED` is set to `true` in the environment variables, Notifications will:
- not process any endpoint type other than `EMAIL_SUBSCRIPTION`
- not allow to create, update, delete, enable or disable an endpoint if its type is not `EMAIL_SUBSCRIPTION`

### CODE REVIEW

**Comment by gwenneg** [comment]
PR check failures are unrelated.

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EmailsOnlyModeTest.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                .basePath(API_INTEGRATIONS_V_1_0)
+                .header(identityHeader)
+                .when()
+                .contentType(JSON)
+                .body(Json.encode(endpoint))
+                .post("/endpoints")
+                .then()
+                .statusCode(400)
+                .extract().asString();
+        assertEquals(UNSUPPORTED_ENDPOINT_TYPE, responseBody);
```
Comment (josejulio) [inline backend/src/test/java/com/redhat/cloud/notifications/routers/EmailsOnlyModeTest.java:65]: if this assert fails, it won't process the `featureFlipper.setEmailsOnlyMode(false);` which might break more tests, making it harder to track it down to the real error.
Could you wrap this into a `try {} finally { featureFlipper.setEmailsOnlyMode(false); }`  or have a `afterEach` setting the mode to `false`?

This applies for all the other tests.

**Review by josejulio — CHANGES_REQUESTED** [review]
Is there any way to disable the `TOCAMEL_CHANNEL` and `FROMCAMEL_CHANNEL` ? 
Else we would need to create them even if we are not using them. Or maybe that's something that can be done already from the ENVs?

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EmailsOnlyModeTest.java (hunk 1 above)
Comment (gwenneg) [reply backend/src/test/java/com/redhat/cloud/notifications/routers/EmailsOnlyModeTest.java]: Good catch, fixed!

**Comment by gwenneg** [comment]
> Is there any way to disable the `TOCAMEL_CHANNEL` and `FROMCAMEL_CHANNEL` ? Else we would need to create them even if we are not using them. Or maybe that's something that can be done already from the ENVs?

Very good question! I added the required param to the ClowdApp template.

**Review by josejulio — APPROVED** [review]

**Comment by gwenneg** [comment]
I renamed the `fromCamel` and `toCamel` channels in the configuration because otherwise the env vars wouldn't work as expected. Now everything works, I tested it locally.

**Pull request merged**